service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseInitiationControllerTest.java

```diff
@@ -263,6 +291,73 @@ void shouldAttemptGrantAccessToAllLocalAuthorityUsersWhenGrantAccessFailsForSome
         postSubmittedEvent(getCase(LA_NOT_IN_PRD_CODE));
 
         verifyGrantCaseRoleAttempts(LA_NOT_IN_PRD_USER_IDS, 2);
+        verifyNoInteractions(caseDataAccessApi);
+    }
+
+    @Test
+    void shouldGrantCaseAssignmentAccessToListOfUsers() {
+        given(featureToggleService.isCaseUserAssignmentEnabled()).willReturn(true);
+        givenPRDWillReturn(LA_IN_PRD_USER_IDS);
+        given(caseDataAccessApi.addCaseUserRoles(any(), any(), any()))
+            .willReturn(AddCaseAssignedUserRolesResponse.builder().status("").build());
+        CallbackRequest callbackRequest = getCase(LA_IN_PRD_CODE);
+        postSubmittedEvent(callbackRequest);
+        AddCaseAssignedUserRolesRequest request = AddCaseAssignedUserRolesRequest.builder()
```

Comment:
you can use
`LA_IN_PRD_USER_IDS.stream().map(userId -> CaseAssignedUserRoleWithOrganisation.builder()
                    .caseRole(LASOLICITOR.formattedName())
                    .caseDataId(callbackRequest.getCaseDetails().getId().toString())
                    .userId(userId)
                    .build()).collect(toList())`

ot you can introduce private method which gives you CaseAssignedUserRoleWithOrganisation for user

Reply:
done .. added 
    private CaseAssignedUserRoleWithOrganisation getCaseAssignedRoleForUser(String caseId, String userId) {
        return CaseAssignedUserRoleWithOrganisation.builder()
            .caseRole(LASOLICITOR.formattedName())
            .caseDataId(caseId)
            .userId(userId)
            .build();
    }